controllers/suite_test.go

```diff
@@ -45,14 +47,93 @@ const (
 )
 
 var (
-	testEnv           *helpers.TestEnvironment
-	clusterReconciler *ClusterReconciler
-	ctx               = context.Background()
+	testEnv  *helpers.TestEnvironment
+	ctx      = context.Background()
+	testEnv1 *helpers.TestEnvironment
```

Comment:
Since there are other ginkgo/envtest tests in this package, I've created a second testEnv for now to isolate the flakes happening in the MHC tests.

- Can we even have two envtest environments running together? 
- Should we have a separate PR to convert the other ginkgo/envtest style tests in this package to use this vanilla go test pattern?